src/query/functions/aggregation/count_values.go

```diff
@@ -257,6 +262,21 @@ func padValuesWithNaNs(vals bucketColumn, size int) bucketColumn {
 	return vals
 }
 
+// IsValid is true if the label name matches the pattern of "^[a-zA-Z_][a-zA-Z0-9_]*$". This
+// method, however, does not use regex for the check but a much faster
+// hardcoded implementation.
+func isValid(ln string) bool {
+	if len(ln) == 0 {
+		return false
+	}
+	for i, b := range ln {
+		if !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || (b >= '0' && b <= '9' && i > 0)) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
nit: maybe this would make more sense as a function in `models`, near where we define `Tags`?